xcm/xcm-builder/src/tests/pay/salary.rs

```diff
@@ -0,0 +1,201 @@
+		CLUB.with(|club| {
+			club.borrow_mut().entry(who.clone()).and_modify(|rank| *rank += 1);
+		});
+		Ok(())
+	}
+	fn demote(who: &Self::AccountId) -> DispatchResult {
+		CLUB.with(|club| match club.borrow().get(who) {
+			None => Err(sp_runtime::DispatchError::Unavailable),
+			Some(&0) => {
+				club.borrow_mut().remove(&who);
+				Ok(())
+			},
+			Some(_) => {
+				club.borrow_mut().entry(who.clone()).and_modify(|rank| *rank += 1);
+				Ok(())
+			},
+		})
+	}
+}
+
+fn set_rank(who: AccountId, rank: u128) {
+	CLUB.with(|club| club.borrow_mut().insert(who, rank));
+}
+
+parameter_types! {
+	pub const RegistrationPeriod: BlockNumber = 2;
+	pub const PayoutPeriod: BlockNumber = 2;
+	pub const FixedSalaryAmount: Balance = 10_000;
+	pub static Budget: Balance = FixedSalaryAmount::get();
+}
+
+pub struct FixedSalary;
+impl GetSalary<Rank, AccountId, Balance> for FixedSalary {
+	fn get_salary(rank: Rank, who: &AccountId) -> Balance {
+		FixedSalaryAmount::get()
+	}
+}
+
+impl pallet_salary::Config for Test {
+	type WeightInfo = ();
+	type RuntimeEvent = RuntimeEvent;
+	type Paymaster = SalaryPayOverXcm;
+	type Members = TestClub;
+	type Salary = FixedSalary;
+	type RegistrationPeriod = RegistrationPeriod;
+	type PayoutPeriod = PayoutPeriod;
+	type Budget = Budget;
+}
+
+fn new_test_ext() -> sp_io::TestExternalities {
+	let t = frame_system::GenesisConfig::default().build_storage::<Test>().unwrap();
+	let mut ext = sp_io::TestExternalities::new(t);
+	ext.execute_with(|| System::set_block_number(1));
+	ext
+}
+
+fn next_block() {
+	System::set_block_number(System::block_number() + 1);
+}
+
+fn run_to(block_number: BlockNumber) {
+	while System::block_number() < block_number {
+		next_block();
+	}
+}
+
+#[test]
+fn salary_pay_over_xcm_works() {
+	let admin_account_id = AccountId::new([0u8; 32]);
+	let account_id = AccountId::new([1u8; 32]);
+	let asset_id: u128 = 100;
+
+	new_test_ext().execute_with(|| {
+		// Create the asset and give enought to the sender
+		assert_ok!(<mock::Assets as Create<_>>::create(
+			asset_id,
+			admin_account_id.clone(),
+			true,
+			FixedSalaryAmount::get() / 2,
+		));
+		assert_ok!(<mock::Assets as Mutate<_>>::mint_into(
+			asset_id,
+			&TreasuryAccountId::get(),
+			FixedSalaryAmount::get() * 2
+		));
+
+		// Set the recipient as a member of a ranked collective
+		set_rank(account_id.clone(), 1);
+
+		// Check starting balance
+		assert_eq!(mock::Assets::balance(AssetIdGeneralIndex::get(), &account_id), 0);
+
+		// Use salary pallet to call `PayOverXcm::pay`
+		assert_ok!(Salary::init(RuntimeOrigin::signed(account_id.clone())));
+		run_to(5);
+		assert_ok!(Salary::induct(RuntimeOrigin::signed(account_id.clone())));
+		assert_ok!(Salary::bump(RuntimeOrigin::signed(account_id.clone())));
+		assert_ok!(Salary::register(RuntimeOrigin::signed(account_id.clone())));
+		run_to(7);
+		assert_ok!(Salary::payout(RuntimeOrigin::signed(account_id.clone())));
+
+		// TODO: Should take the message sent by the salary pallet.
```

Comment:
Any ideas how I could do this?

Reply:
Yes, specifically the line that's commented below, that doesn't work because of types.

```rust
let (_, message, hash) = sent_xcm().first().unwrap();
```